rollup-interface/src/state_machine/stf.rs

```diff
@@ -99,15 +102,19 @@ pub trait StateTransitionFunction<Vm: Zkvm, B: BlobReaderTrait> {
 
     /// The contents of a transaction receipt. This is the data that is persisted in the database
     type TxReceiptContents: Serialize + DeserializeOwned + Clone;
+
     /// The contents of a batch receipt. This is the data that is persisted in the database
     type BatchReceiptContents: Serialize + DeserializeOwned + Clone;
 
     /// Witness is a data that is produced during actual batch execution
     /// or validated together with proof during verification
     type Witness: Default + Serialize;
 
+    /// The validity condition that must be verified outside of the Vm
+    type Condition: ValidityCondition;
+
     /// Perform one-time initialization for the genesis block.
-    fn init_chain(&mut self, params: Self::InitialState);
+    fn init_chain(&mut self, params: Self::InitialState) -> anyhow::Result<[u8; 32]>;
```

Comment:
Can we add docs indicating what is returned?

Also, shouldn't type be `thiserror` ?

Reply:
I think anyhow works because this function propagates error from sub function calls (genesis and validate_and_commit)